AC3.2-MidUnit-6-Exam/DoubleVideoViewController.swift

```diff
@@ -7,25 +7,128 @@
     
+    //MARK: - UIImagePickerControllerDelegate
+    func imagePickerController(_ picker: UIImagePickerController, didFinishPickingMediaWithInfo info: [String : Any]) {
+        switch info[UIImagePickerControllerMediaType] as! String {
+        case String(kUTTypeMovie):
+            if let url = info[UIImagePickerControllerReferenceURL] as? URL {
+                self.movieUrl = url
+                
+            }
+        case String(kUTTypeImage):
+            
+            if (info[UIImagePickerControllerOriginalImage] as? UIImage) != nil {
+                dismiss(animated: true, completion: nil)
+                
+                let alertController = UIAlertController(title: "No Images", message: "Select a Video", preferredStyle: .alert)
+                let cancel = UIAlertAction(title: "Ok", style: .cancel, handler: nil)
+                alertController.addAction(cancel)
+                present(alertController, animated: true)
+                
+            }
+        default:
+            print("Not Complatable Media")
+        }
+        
+        dismiss(animated: true, completion: nil)
+        
+        if let url = self.movieUrl {
+            if player?.rate == nil || player?.rate == 0.0 {
+            
+            let item = AVPlayerItem(url: url)
+             player = AVPlayer(playerItem: item)
+            
+            let playerLayer = AVPlayerLayer(player: player)
+        
+                self.videoContainerTop.layer.addSublayer(playerLayer)
+                player?.playImmediately(atRate: 1.0)
+                playerLayer.frame = self.videoContainerTop.bounds
+               
+            } else if secondPlayer?.rate == nil || secondPlayer?.rate == 0.0 {
+                
+                let item = AVPlayerItem(url: url)
+                secondPlayer = AVPlayer(playerItem: item)
+                
+                let playerLayer = AVPlayerLayer(player: secondPlayer)
+                
+                self.videoContainerBottom.layer.addSublayer(playerLayer)
+                secondPlayer?.playImmediately(atRate: 1.0)
+                playerLayer.frame = self.videoContainerBottom.bounds
+                
+            } else {
+             
+                let alertController = UIAlertController(title: "Both Positions are playing Videos", message: "Cant play another video at this moment", preferredStyle: .alert)
+                let cancel = UIAlertAction(title: "Ok", style: .cancel, handler: nil)
+                alertController.addAction(cancel)
+                present(alertController, animated: true)
+            }
+            
+            
+        }
+        
+    }
+
+    
+    
+    @IBAction func loadVideo(_ sender: UIButton) {
+        let imagePickerController = UIImagePickerController()
+        imagePickerController.allowsEditing = false
+        imagePickerController.sourceType = .photoLibrary
+        imagePickerController.mediaTypes = [ String(kUTTypeMovie), String(kUTTypeImage) ]
```

Comment:
You could have limited this to just movies by leaving images out of ```mediaTypes```. You do handle images with an alert which is good. No points off but this might have put some of the tiny mark down in style/quality.